pkg/manager/member/tikv_group_member_manager.go

```diff
@@ -92,3 +145,529 @@ func (tgm *tikvGroupMemberManager) registerTiKVGroup(tg *v1alpha1.TiKVGroup, tc
+		if store.Store != nil && !pattern.Match([]byte(store.Store.Address)) {
+			continue
+		}
+		status := getTiKVStore(store)
+		if status == nil {
+			continue
+		}
+		tombstoneStores[status.ID] = *status
+	}
+
+	tg.Status.Synced = true
+	tg.Status.Stores = stores
+	tg.Status.TombstoneStores = tombstoneStores
+	tg.Status.Image = ""
+	c := filterContainer(set, "tikv")
+	if c != nil {
+		tc.Status.TiKV.Image = c.Image
+	}
+	return nil
+}
+
+func (tgm *tikvGroupMemberManager) tikvGroupStatefulSetIsUpgrading(tg *v1alpha1.TiKVGroup, set *apps.StatefulSet) (bool, error) {
+	if statefulSetIsUpgrading(set) {
+		return true, nil
+	}
+	selector, err := label.NewGroup().Instance(tg.Name).TiKV().Selector()
+	if err != nil {
+		return false, err
+	}
+	tikvPods, err := tgm.podLister.Pods(set.Namespace).List(selector)
+	if err != nil {
+		return false, err
+	}
+	for _, pod := range tikvPods {
+		revisionHash, exist := pod.Labels[apps.ControllerRevisionHashLabelKey]
+		if !exist {
+			return false, nil
+		}
+		if revisionHash != tg.Status.StatefulSet.UpdateRevision {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// TODO: add unit test
+func newServiceForTiKVGroup(tg *v1alpha1.TiKVGroup, svcName string) *corev1.Service {
+	ns := tg.Namespace
+	tgName := tg.Name
+	tgLabel := label.NewGroup().TiKV().Instance(tgName)
+	svcLabel := label.NewGroup().TiKV().Instance(tgName).UsedByPeer()
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      svcName,
+			Namespace: ns,
+			Labels:    svcLabel.Labels(),
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "peer",
+					Port:       20160,
+					TargetPort: intstr.FromInt(20160),
+					Protocol:   corev1.ProtocolTCP,
+				},
+			},
+			Selector:                 tgLabel.Labels(),
+			PublishNotReadyAddresses: true,
+			ClusterIP:                "None",
+		},
+	}
+	return svc
+}
+
+// TODO: add unit test
+func getTikVConfigMapForTiKVGroup(tg *v1alpha1.TiKVGroup, tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
+	if tg.Spec.Config == nil {
+		tg.Spec.Config = &v1alpha1.TiKVConfig{}
+	}
+	config := tg.Spec.Config
+	if tc.IsTLSClusterEnabled() {
+		if config.Security == nil {
+			config.Security = &v1alpha1.TiKVSecurityConfig{}
+		}
+		config.Security.CAPath = pointer.StringPtr(path.Join(tikvClusterCertPath, tlsSecretRootCAKey))
+		config.Security.CertPath = pointer.StringPtr(path.Join(tikvClusterCertPath, corev1.TLSCertKey))
+		config.Security.KeyPath = pointer.StringPtr(path.Join(tikvClusterCertPath, corev1.TLSPrivateKeyKey))
+	}
+	confText, err := MarshalTOML(config)
+	if err != nil {
+		return nil, err
+	}
+	scriptModel := &TiKVStartScriptModel{
+		Scheme:                    tc.Scheme(),
+		EnableAdvertiseStatusAddr: false,
+		DataDir:                   filepath.Join(tikvDataVolumeMountPath, tc.Spec.TiKV.DataSubDir),
+	}
+	if tc.Spec.EnableDynamicConfiguration != nil && *tc.Spec.EnableDynamicConfiguration {
```

Comment:
This was added to avoid rolling update during operator upgrade from the releases that do not support this feature, but for the new TiKV group, I think it's ok to set `EnableAdvertiseStatusAddr` to `true` directly, no need to add the dependency to the tc.

Reply:
updated.